### PR TITLE
dropbear: bump to 2022.82

### PR DESCRIPTION
**Pull request by rockdrilla**
dropbear: bump to 2022.82

- update dropbear to latest stable 2022.82;
  for the changes see https://matt.ucc.asn.au/dropbear/CHANGES
- use $(AUTORELEASE) in PKG_RELEASE
- use https for all uris
- refresh all patches
- rewrite patches:
  - 100-pubkey_path.patch
  - 130-ssh_ignore_x_args.patch

binary/pkg size changes:
- ath79/generic, mips:
  - binary: 215112 -> 219228 (+4116)
  - pkg: 111914 -> 113404 (+1490)
- ath79/tiny, mips:
  - binary: 172501 -> 172485 (-16)
  - pkg: 89871 -> 90904 (+1033)

Signed-off-by: Konstantin Demin <rockdrilla@gmail.com>

### CODE REVIEW

**Comment by rockdrilla** [comment]
binary/package size diffs are coming soon.

**Comment by rockdrilla** [comment]
Run-tested on ath79/generic (Archer C7 v2).

**Comment by Borromini** [comment]
Run-tested on mvebu/cortexa72 (RB5009UG+S+IN) and ramips/mt7621 (EAP235-Wall v1).

Tested-by: Stijn Segers <foss@volatilesystems.org>

File: package/network/services/dropbear/Makefile [hunk 1]
```diff
@@ -130,8 +130,10 @@ DB_OPT_COMMON = \
 DB_OPT_CONFIG = \
 	DROPBEAR_CURVE25519|CONFIG_DROPBEAR_CURVE25519|1|0 \
 	DROPBEAR_ED25519|CONFIG_DROPBEAR_ED25519|1|0 \
+	DROPBEAR_SK_ED25519|CONFIG_DROPBEAR_ED25519|1|0 \
 	DROPBEAR_CHACHA20POLY1305|CONFIG_DROPBEAR_CHACHA20POLY1305|1|0 \
 	DROPBEAR_ECDSA|CONFIG_DROPBEAR_ECC|1|0 \
+	DROPBEAR_SK_ECDSA||CONFIG_DROPBEAR_ECC|1|0 \
```
Comment (rsalvaterra) [inline package/network/services/dropbear/Makefile:136]: Double ||, is this correct? The rest looks good to me.

**Comment by rockdrilla** [comment]
@rsalvaterra, thanks for pointing out typo! :)

**Comment by rsalvaterra** [comment]
> @rsalvaterra, thanks for pointing out typo! :)

You're welcome! Looks good to me now, thanks.

**Comment by rockdrilla** [comment]
@Borromini, I was a little confused from this comment:
> There's little difference with master at this point.

**Comment by Borromini** [comment]
> @Borromini, I was a little confused from this comment:
> 
> > There's little difference with master at this point.

What I meant to say was that the 22.03 codebase is very similar to master at this point still (dependencies etc.), so I believe testing on 22.03 is as 'valid' as if I were to test it on master at this time. Sorry for the confusion.

**Comment by hauke** [comment]
Thank you for your patch, I applied it to master in 65256aee23a5104eb0c78411fdc73640c0b757ea.

**Pull request closed without merging**